src/main/scala/de/upb/cs/uc4/hyperledger/utilities/EnrollmentManager.scala

```diff
@@ -13,31 +13,31 @@ object EnrollmentManager {
     val wallet = WalletManager.getWallet(walletPath)
 
     // check if user already exists in my wallet
-    if (wallet.list.contains(username)) {
-      println(s"An identity for the user $username already exists in the wallet.")
+    if (WalletManager.containsIdentity(wallet, username)) {
+      println(s"[DEBUG] :: An identity for the user $username already exists in the wallet.")
     }
     else {
-      println(s"Try to get the identity for the user $username.")
-
-      // Create a CA client for interacting with the CA
-      val props = new Properties
-      props.put("pemFile", tlsCert.toAbsolutePath.toString)
-      props.put("allowAllHostNames", "true")
-      val caClient = HFCAClient.createNewInstance(caURL, props)
-      val cryptoSuite = CryptoSuiteFactory.getDefault.getCryptoSuite
-      caClient.setCryptoSuite(cryptoSuite)
-      println(s"Created cryptoSuite.")
-
-      // enroll my user
-      val enrollmentRequestTLS = new EnrollmentRequest
-      enrollmentRequestTLS.addHost("localhost")
-      enrollmentRequestTLS.setProfile("tls")
+      println(s"[DEBUG] :: Try to get the identity for the user $username.")
+
+      val caClient = CAClientManager.getCAClient(caURL, tlsCert)
+      println("[DEBUG] :: Retrieved CAClient")
+
+      val enrollmentRequestTLS = EnrollmentManager.prepareEnrollmentRequest("localhost", "tls")

```

Comment:
Not sure about addressing in kubernetes (and if peers check hostnames at all), but if we have different organizations running peers in different clusters, the host should probably be configurable, right?

Reply:
created an issue

https://github.com/upb-uc4/hlf-api/issues/38